tests/libpod/container.go

```diff
@@ -41,3 +43,16 @@ func LookupContainer(pod *v1.Pod, containerName string) *v1.Container {
 	}
 	panic(fmt.Errorf("could not find the %s container", containerName))
 }
+
+func LookupComputeContainerFromVmi(vmi *virtv1.VirtualMachineInstance) (*v1.Container, error) {
```

Comment:
imho this function is a bit cumbersome to call, and forces this library to import a new dependency. There's something fishy if a library such as `libpod` is exposing both two low-level functions (`LookupComputeContainer`, `GetPodByVirtualMachineInstance`) as well as their composition (your proposed function).


How do you feel about a simpler replace of

`GetVmiPod(virtClient, vmi)`

with

`LookupComputeContainer(GetPodByVirtualMachineInstance(vmi))` ?


BTW, in a different commit, I hope we can eliminate the `namespace` argument of `GetPodByVirtualMachineInstance`.

Reply:
TBH, I tend to disagree.

As a general opinion, I think that it's valid and even preferred to combine two function calls into one if the two function calls are very frequently being used together. As my commit shows, **all usages** of `GetVmiPod()` under `tests/hotplug/` are there just to provide the right argument to `LookupComputeContainer()`.

I understand that in many (or even most) cases it's better, as a general rule, to separate function calls in a fine-grained matter. But this general rule is not sacred. Practically, calling:
`LookupComputeContainer(GetPodByVirtualMachineInstance(vmi))`

is MUCH more cumbersome to call then `LookupComputeContainerFromVmi(vmi)`.

These functions are used together and will continue to be used together for the unforeseen future. I think providing a form of "syntactic sugar" here has pros without any practical cons as far as I can tell.